app/src/main/java/de/westnordost/streetcomplete/util/CrashReportExceptionHandler.kt

```diff
@@ -54,12 +59,16 @@ class CrashReportExceptionHandler(
     }
 
     private fun askUserToSendErrorReport(activityCtx: Activity, @StringRes titleResourceId: Int, error: String?) {
+        val logText = readLogFromDatabase()
```

Comment:
There is a slight problem with this implementation.

On crash, the crash report (stack trace, android version etc.) are all written to a file. Then, the next time the user opens the app, which might very well be a week after, he is asked to send the error. But by this time, the log may not actually exist in the database anymore.
Furthermore, (another small point,) these functions are executed on the UI thread.

So, how about this:
- Log is fetched from the database table and written into the crash report file at the time of the crash
- For `askUserToSendErrorReport(activityCtx: Activity, @StringRes titleResourceId: Int, e: Exception)`, either pass the log lines as parameter or retrieve them automatically in that function

Also, I note now that I look at the file, and it is not your fault, that the error reporting is inconsistent: A full text wth app version, device name etc. is attached on an uncaught exception, but when `askUserToSendErrorReport` is called from application code, it is not. Right? Or am I misreading something?

Reply:
> Right? Or am I misreading something?

Right. On errors during download or upload, `askUserToSendErrorReport` is called and results in a report without device details etc. (very recently noticed that in SCEE)

Reply:
Solved both issues by extracting creation of a full report text to a separate `createErrorReport` function, so both uncaught and caught errors generate similar report with app version, logs, etc.